rest-service/manager_rest/resource_manager.py

```diff
@@ -482,15 +482,37 @@ def delete_deployment(self,
         deployment = self.sm.get(models.Deployment, deployment_id)
 
         # Validate there are no running executions for this deployment
-        deplyment_id_filter = self.create_filters_dict(
+        deployment_id_filter = self.create_filters_dict(
             deployment_id=deployment_id,
             status=ExecutionState.ACTIVE_STATES + ExecutionState.QUEUED_STATE
         )
         executions = self.sm.list(
             models.Execution,
-            filters=deplyment_id_filter
+            filters=deployment_id_filter
         )
 
+        # Verify deleting the deployment won't affect dependent deployments
+        dep_graph = RecursiveDeploymentDependencies(get_storage_manager())
+        excluded_id = self._excluded_component_creator_id(deployment)
+        deployment_dependencies = \
+            dep_graph.retrieve_and_display_dependencies(
+                deployment.id,
+                excluded_component_creator_id=excluded_id)
+        if deployment_dependencies:
+            if ignore_live_nodes:
```

Comment:
Maybe we should change the var name: `ignore_live_nodes` to something like `force_delete` or anything else? I guess `ignore_live_nodes` refer to the fact that the deployment itself has live nodes that it deployed.